plugins/destination/gcs/client/client_test.go

```diff
@@ -120,7 +139,13 @@ func testPluginCustom(t *testing.T, s *spec.Spec) {
 
 	readRecords, err := readAll(ctx, client, table)
 	if err != nil {
+		if expectErrorWhenReading {
+			assert.ErrorContains(t, err, readErrorString)
+			t.Skip()
+		}
 		t.Fatal(fmt.Errorf("failed to sync: %w", err))
+	} else if expectErrorWhenReading {
+		t.Fatal(fmt.Errorf("expected error while performing client.read(), got None"))
 	}
```

Comment:
```suggestion
	if expectErrorWhenReading {
		require.ErrorContains(t, err,readErrorString)
	} else {
		require.NoError(t, err)
	}
```

Could this work instead? Regardless what's the reason we have `t.Skip()` in the test?

Reply:
Yes, that works much better, thanks @erezrokah. This is my first time using Go, so this is much appreciated! 

I was using t.Skip() to skip the rest of the test when we encounter an error but I saw [this file](https://github.com/cloudquery/cloudquery/blob/7a0936dcb17412460cce67e2f7733db2261fb892/cli/internal/publish/images/images_test.go#L270) uses return, so I changed it to that.

Reply:
OK so I think we can drop the `return` in both cases. That's the reason to use `require` over `assert`. `require` stops the test on failure https://github.com/stretchr/testify?tab=readme-ov-file#require-package

Reply:
@erezrokah I think we still need this return, require will stop the tests on failure, but failure in this case would mean either not getting an error or the error not containing the string that we provided. If the error contains the message, the `require` assertion would pass successfully and move on to the next part of the tests

Reply:
You're right 👍 How about https://github.com/cloudquery/cloudquery/pull/17602/commits/a25ccda535682d27c6e322d9668e8866acd98086 to early return and reduce the `if/else` nesting?

I also added another commit to remove the boolean flag in https://github.com/cloudquery/cloudquery/pull/17602/commits/0982124946b8b6a767963c5d01bcfba490f0fe6a

WDYT?

Reply:
Yes, much cleaner, thanks!